testcases/OpTestDlparIO.py

```diff
@@ -0,0 +1,289 @@
+                       dlpar move" % self.dest_lpar)
+
+        # dlpar move operation from lpar2 to lpar1
+        self.changehwres(self.mg_system, 'm', self.dest_lpar_id, self.cv_HMC.lpar_name,
+                         self.drc_index, 'move')
+        output = self.listhwres(self.mg_system, self.cv_HMC.lpar_name, self.drc_index)
+        if not output:
+            log.debug(output)
+            raise OpTestError("lshwres fails to list the drc in lpar %s after \
+                       dlpar move" % self.cv_HMC.lpar_name)
+        output = self.listhwres(self.server, self.dest_lpar, self.drc_index)
+        if output:
+            log.debug(output)
+            raise OpTestError("lshwres still lists the drc in dest lpar %s after \
+                      dlpar move to source lpar %s" % (self.dest_lpar, self.cv_HMC.lpar_name))
+
+    def listhwres(self, server, lpar, drc_index):
+        cmd = 'lshwres -r io -m %s \
+               --rsubtype slot --filter lpar_names= %s \
+               | grep -i %s' % (server, lpar, drc_index)
+        return self.cv_HMC.ssh.run_command(cmd)
+
+    def listhwres_sriov(self, server, lpar, logical_port_id):
+        cmd = 'lshwres -r sriov -m %s \
+              --rsubtype logport --filter lpar_names= %s --level eth \
+              | grep -i %s' % (server, lpar, logical_port_id)
+        return self.cv_HMC.ssh.run_command(cmd)
+
+    def changehwres(self, server, operation, lpar_id, lpar, drc_index, msg):
+        if operation == 'm':
+            cmd = 'chhwres -r io --rsubtype slot -m %s \
+               -o %s --id %s -t %s -l %s ' % (server, operation, lpar_id,
+                                              lpar, drc_index)
+        else:
+            cmd = 'chhwres -r io --rsubtype slot -m %s \
+                   -o %s --id %s -l %s ' % (server, operation, lpar_id,
+                                            drc_index)
+        self.cv_HMC.ssh.run_command(cmd)
+
+    def changehwres_sriov(self, server, operation, lpar_id, adapter_id,
+                          logical_port_id, phys_port_id, msg):
+        if operation == 'r':
+            cmd = 'chhwres -r sriov -m %s --rsubtype logport -o r --id %s -a \
+                  adapter_id=%s,logical_port_id=%s' \
+                  % (server, lpar_id, adapter_id, logical_port_id)
+        elif operation == 'a':
+            cmd = 'chhwres -r sriov -m %s --rsubtype logport -o a --id %s -a \
+                  phys_port_id=%s,adapter_id=%s,logical_port_id=%s, \
+                  logical_port_type=eth' % (server, lpar_id, phys_port_id,
+                                            adapter_id, logical_port_id)
+        self.cv_HMC.ssh.run_command(cmd)
+
+    def runTest(self):
+        '''
+        DLPAR remove, add and move operations from lpar_1 to lpar_2
+        '''
+        for _ in range(self.num_of_dlpar):
+            self.dlpar_remove()
+            self.dlpar_add()
+            #self.dlpar_move()
```

Comment:
This is commented, meaning it is not yet supported ?

Reply:
I need to discuss with Narashiman on the same, I see a conflict here as it was migrated from avocado tests.